contrib/epee/include/net/http_auth.h

```diff
@@ -71,8 +71,8 @@ namespace net_utils
         std::uint32_t counter;
       };
 
-      http_server_auth() : user() {}
-      http_server_auth(login credentials);
+      http_server_auth(std::function<void(size_t, uint8_t*)> r) : user(), rng(r) {}
```

Comment:
This can remain a default constructor (i.e. leave `rng` in default state). Its not possible to access the `rng(...)` call currently unless `user` is also set. This check occurs just a few lines below this one.

Reply:
This is less maintainable though, isn't it ? Since rng is sometimes set and sometimes not, which was the point of the request to change.

Reply:
done anyway

Reply:
Since my requests seem arbitrary, I'll explain a bit further.

My recommendations are based on the observation that invariants in code are most likely to be noticed within a function, then class, then file, and then finally across files. It is very easy to "move" a constructor call in another file without realizing there is a dependency on another function call. Typically I will only have an invariant within a single function or class, then place `assert(...)` in the code as a "signal" to future maintainers that there is an important invariant being checked or maintained _somewhere_ else. This really isn't possible across files because there is no expectation that a programmer would look at the header or source to simply move the construction of an object.

However, you may have had a point. It is somewhat unlikely that someone will drop the call to initialize the RNG, or construct + use the http auth object before calling initialization. And even if this did happen, `std::function` throws in that situation, so the behavior **is** defined. A simple comment in the header listing this dependency may have been sufficient. My apologies for annoyances - the changes seemed pretty minimal which is why I recommended it.